tests/ops/op_math/test_adjoint_op.py

```diff
@@ -0,0 +1,587 @@
+        assert op.name == "Adjoint(T.inv)"
+
+
+class TestAdjointOperationDiffInfo:
+    """Test differention related properties and methods of AdjointOperation."""
+
+    def test_grad_method_None(self):
+        """Test grad_method copies base grad_method when it is None."""
+        base = qml.PauliX(0)
+        op = Adjoint(base)
+
+        assert op.grad_method is None
+
+    @pytest.mark.parametrize("op", (qml.RX(1.2, wires=0),))
+    def test_grad_method_not_None(self, op):
+        """Make sure the grad_method property of a Adjoint op is the same as the base op."""
+        assert Adjoint(op).grad_method == op.grad_method
+
+    @pytest.mark.parametrize(
+        "base", (qml.PauliX(0), qml.RX(1.234, wires=0), qml.Rotation(1.234, wires=0))
+    )
+    def test_grad_recipe(self, base):
+        """Test that the grad_recipe of the Adjoint is the same as the grad_recipe of the base."""
+        assert Adjoint(base).grad_recipe == base.grad_recipe
+
+    def test_get_parameter_shift(self):
+        """Test `get_parameter_shift` for an operation where it still doesn't raise warnings and errors."""
+        base = qml.Rotation(1.234, wires=0)
+        with pytest.warns(UserWarning, match=r"get_parameter_shift is deprecated."):
+            assert Adjoint(base).get_parameter_shift(0) == base.get_parameter_shift(0)
+
+    @pytest.mark.parametrize(
+        "base",
+        (qml.RX(1.23, wires=0), qml.Rot(1.23, 2.345, 3.456, wires=0), qml.CRX(1.234, wires=(0, 1))),
+    )
+    def test_parameter_frequencies(self, base):
+        """Test that the parameter frequencies of an Adjoint are the same as the parameter-frequencies of the base."""
+        assert Adjoint(base).parameter_frequencies == base.parameter_frequencies
+
+
+class TestQueueing:
+    """Test that Adjoint operators queue and update base metadata"""
+
+    def test_queueing(self):
+        """Test queuing and metadata when both Adjoint and base defined inside a recording context."""
+
+        with qml.tape.QuantumTape() as tape:
+            base = qml.Rot(1.2345, 2.3456, 3.4567, wires="b")
+            op = Adjoint(base)
+
+        assert tape._queue[base]["owner"] is op
+        assert tape._queue[op]["owns"] is base
+        assert tape.operations == [op]
+
+    def test_queueing_base_defined_outside(self):
+        """Test that base is added to queue even if it's defined outside the recording context."""
+
+        base = qml.Rot(1.2345, 2.3456, 3.4567, wires="b")
+        with qml.tape.QuantumTape() as tape:
+            op = Adjoint(base)
+
+        assert tape._queue[base]["owner"] is op
+        assert tape._queue[op]["owns"] is base
+        assert tape.operations == [op]
+
+    def test_do_queue_False(self):
+        """Test that when `do_queue` is specified, the operation is not queued."""
+        base = qml.PauliX(0)
+        with qml.tape.QuantumTape() as tape:
+            op = Adjoint(base, do_queue=False)
+
+        assert len(tape) == 0
```

Comment:
Nice to have this in, even if we do not know yet what happens with `do_queue`

Reply:
Much easier to remove a test when a feature is removed than figure out what's going wrong when a feature is changed.